### PR TITLE
Removing kernel stack watchdog debug build, some other fixes

### PR DESCRIPTION
**Pull request by anirbanr-fb**
Summary:

1. comment out different instances of kernel stack watchdog
   activity by making timedelay = 0

2. support for debug build

3. replace DCHECK_NOTNULL with DCHECK's where it would fail
4. Fixing the issue with MANAGER_INITIALIZED state
5. Making Log::Init a virtual function, otherwise we will
   potentially exercise unsafe code in Wrapper mode.

Test Plan: tp2_use_local and ran the mtr test

Reviewers: iRitwik, bhatvinay

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: cmake_modules/FindLz4.cmake [hunk 1]
```diff
@@ -25,7 +25,7 @@ find_path(LZ4_INCLUDE_DIR lz4.h
   # make sure we don't accidentally pick up a different version
   NO_CMAKE_SYSTEM_PATH
   NO_SYSTEM_ENVIRONMENT_PATH)
-find_library(LZ4_STATIC_LIB liblz4.a
+find_library(LZ4_STATIC_LIB NAMES liblz4_pic.a liblz4.a
```
Comment (iRitwik) [inline cmake_modules/FindLz4.cmake:28]: What does the _pic library do?

File: src/kudu/consensus/consensus_peers.cc [hunk 2]
```diff
@@ -503,8 +503,10 @@ Peer::~Peer() {
 
 RpcPeerProxy::RpcPeerProxy(gscoped_ptr<HostPort> hostport,
                            gscoped_ptr<ConsensusServiceProxy> consensus_proxy)
-    : hostport_(std::move(DCHECK_NOTNULL(hostport))),
-      consensus_proxy_(std::move(DCHECK_NOTNULL(consensus_proxy))) {
+    : hostport_(std::move(hostport)),
```
Comment (iRitwik) [inline src/kudu/consensus/consensus_peers.cc:506]: Why does moving the null check inside the body beneficial?

File: src/kudu/util/thread.cc [hunk 3]
```diff
@@ -552,7 +552,7 @@ Status Thread::StartThread(const std::string& category, const std::string& name,
 
   {
     SCOPED_LOG_SLOW_EXECUTION_PREFIX(WARNING, 500 /* ms */, log_prefix, "creating pthread");
-    SCOPED_WATCH_STACK((flags & NO_STACK_WATCHDOG) ? 0 : 250);
+    //SCOPED_WATCH_STACK((flags & NO_STACK_WATCHDOG) ? 0 : 250);
```
Comment (iRitwik) [inline src/kudu/util/thread.cc:555]: Did you run into any perf issues with the watchdog?
They might have fixed it in a later revision: http://mail-archives.eu.apache.org/mod_mbox/kudu-commits/201610.mbox/%3C16d2131cc82e47f0ab4638829cca1d54@git.apache.org%3E

**Review by iRitwik — APPROVED** [review]
I have accepted with a few questions.

File: cmake_modules/FindLz4.cmake (hunk 1 above)
Comment (anirbanr-fb) [reply cmake_modules/FindLz4.cmake:28]: pic library has position independent code generation. We have 2 libraries _pic.a and .a for each tp2 library. When we compile debug mode kuduraft, it complains of unhandle-able relocations, which essentially means that you need to link with PIC code.

File: src/kudu/consensus/consensus_peers.cc (hunk 2 above)
Comment (anirbanr-fb) [reply src/kudu/consensus/consensus_peers.cc:506]: DCHECK_NOTNULL does not compile any more. For gscoped ptr, it bombs.
I am trying to not lose DEBUG checks while making the code compile in pure debug mode.

File: src/kudu/util/thread.cc (hunk 3 above)
Comment (anirbanr-fb) [reply src/kudu/util/thread.cc:555]: Kernel Stack Watchdog thread was causing the crashes in uninstall plugin path. The RPC worker threads and supervisor threads would crash intermittently due to this feature.

**Pull request merged**